flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplit.java

```diff
@@ -57,38 +69,64 @@ public boolean equals(Object o) {
             return false;
         }
         HybridSourceSplit that = (HybridSourceSplit) o;
-        return sourceIndex == that.sourceIndex && wrappedSplit.equals(that.wrappedSplit);
+        return sourceIndex == that.sourceIndex
+                && Arrays.equals(wrappedSplitBytes, that.wrappedSplitBytes);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(wrappedSplit, sourceIndex);
+        return Objects.hash(wrappedSplitBytes, sourceIndex);
     }
 
     @Override
     public String toString() {
         return "HybridSourceSplit{"
                 + "realSplit="
-                + wrappedSplit
+                + wrappedSplitBytes
                 + ", sourceIndex="
                 + sourceIndex
```

Comment:
Add the `splitId` field.

Reply:
Done. I also removed `wrappedSplitBytes` because it doesn't provide meaningful information.